### PR TITLE
fix: use random_get instead of arc4random as arc4random does not work correctly with wizer

### PR DESCRIPTION
**Pull request by JakeChampion**
wizer causes the seed in arc4random to be the same between executions which is not random

### CODE REVIEW

File: c-dependencies/js-compute-runtime/js-compute-builtins.cpp [hunk 1]
```diff
@@ -4402,6 +4403,18 @@ bool process_network_io(JSContext *cx) {
   return true;
 }
 
+bool math_random(JSContext *cx, unsigned argc, Value *vp) {
+  int32_t storage;
+  int32_t *buf = &storage;
+  random_get((int32_t)buf, 8);
+  uint32_t value = std::abs(storage);
+  double newvalue = static_cast<float>(value) / std::powf(2.0, 32.0);
```
Comment (elliottt) [inline c-dependencies/js-compute-runtime/js-compute-builtins.cpp:4411]: I'd probably do this all as a `double` if the result will be used as a double.
```suggestion
  double newvalue = static_cast<double>(value) / std::pow(2.0, 32.0);
```

File: c-dependencies/js-compute-runtime/builtins/crypto.cpp [hunk 2]
```diff
@@ -44,7 +45,7 @@ bool Crypto::get_random_values(JSContext *cx, unsigned argc, JS::Value *vp) {
   JS::AutoCheckCannotGC noGC(cx);
   bool is_shared;
   void *buffer = JS_GetArrayBufferViewData(typed_array, &is_shared, noGC);
-  arc4random_buf(buffer, byte_length);
+  random_get((int32_t)buffer, byte_length);
```
Comment (elliottt) [inline c-dependencies/js-compute-runtime/builtins/crypto.cpp:48]: Isn't this going to be pretty slow? Should we pull over a PRNG and seed it from `random_get` when initializing?

File: tests/wpt-harness/expectations/fetch/api/basic/request-upload.any.js.json [hunk 3]
```diff
@@ -48,7 +48,7 @@
     "status": "FAIL"
   },
   "Fetch with POST with text body on 421 response should be retried once on new connection.": {
-    "status": "PASS"
+    "status": "FAIL"
```
Comment (elliottt) [inline tests/wpt-harness/expectations/fetch/api/basic/request-upload.any.js.json:51]: Do you know why this test is failing now?

**Review by elliottt — COMMENTED** [review]
Looking good, just a few questions :)

File: tests/wpt-harness/expectations/fetch/api/basic/request-upload.any.js.json (hunk 3 above)
Comment (JakeChampion) [reply tests/wpt-harness/expectations/fetch/api/basic/request-upload.any.js.json:51]: The more surprising result was that it ever reported as passing, our fetch implementation does not include retry logic - which is what this test is attempting to exercise

Comment (elliottt) [reply tests/wpt-harness/expectations/fetch/api/basic/request-upload.any.js.json:51]: Quite mysterious!

**Pull request closed without merging**